core/src/main/scala/com/criteo/cuttle/ThreadPools.scala

```diff
@@ -13,8 +14,8 @@ object ThreadPools {
   }
 
   sealed trait Metrics {
-    protected var _threadPoolSize: Long = 0
-    def threadPoolSize(): Long = _threadPoolSize
+    protected var _threadPoolSize: AtomicInteger = new AtomicInteger(0)
```

Comment:
no need for var here. 
Also, I think thread naming and counting is definitely an improvement, but I would like to avoid having a shared mutable thread counter. I'd rather leave it an implementaton detail of the ThreadPoolFactory

Reply:
Oops yes indeed about the var I must have forgotten to change it back after wrapping the POD inside an Atomic object.
Sure I'll fix this it makes more sense as you said.